### ISSUE TITLE
Is it worth trying to run this in parallel?

### ISSUE THREAD

**Issue by RustyRaptor** [hashcat/maskprocessor#38]
I noticed that it is strictly single threaded. I was wondering does it make sense to try to split it up into multiple instances running at different points? Would this actually improve performance? Anyone tried it?

**Comment by RustyRaptor** [comment]
Is this the actual purpose of the function to start and stop at different words?

**Comment by roycewilliams** [comment]
The purpose is to generate words/strings. Often, if someone is concerned about maskprocessor performance enough to want to significantly parallelize it, they're doing some kind of attack that could be better served by doing candidate generation within the cracking software itself (like using masks within hashcat).

https://hashcat.net/wiki/doku.php?id=maskprocessor

**Comment by RustyRaptor** [comment]
Thanks. I'm pretty sure that's exactly what is needed in this case
